src/test/java/seedu/address/storage/JsonAdaptedInternshipTest.java

```diff
@@ -0,0 +1,159 @@
+                "The id field must not be negative.",
+                jsonAdaptedInternship.get()::toModelType);
+    }
+
+    @Test
+    public void loadToModel_missingRole_failure() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("missingRoleInternship.json"),
+                        JsonAdaptedInternship.class);
+        assertThrows(IllegalValueException.class,
+                String.format(MISSING_FIELD_MESSAGE_FORMAT, "role"),
+                jsonAdaptedInternship.get()::toModelType);
+    }
+
+    @Test
+    public void loadToModel_missingFrom_failure() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("missingFromInternship.json"),
+                        JsonAdaptedInternship.class);
+        assertThrows(IllegalValueException.class,
+                String.format(MISSING_FIELD_MESSAGE_FORMAT, Time.class.getSimpleName()),
+                jsonAdaptedInternship.get()::toModelType);
+    }
+
+    @Test
+    public void loadToModel_invalidFrom_failure() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("invalidFromInternship.json"),
+                        JsonAdaptedInternship.class);
+        assertThrows(IllegalValueException.class,
+                Time.MESSAGE_CONSTRAINTS,
+                jsonAdaptedInternship.get()::toModelType);
+    }
+
+    @Test
+    public void loadToModel_missingTo_failure() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("missingToInternship.json"),
+                        JsonAdaptedInternship.class);
+        assertThrows(IllegalValueException.class,
+                String.format(MISSING_FIELD_MESSAGE_FORMAT, Time.class.getSimpleName()),
+                jsonAdaptedInternship.get()::toModelType);
+    }
+
+    @Test
+    public void loadToModel_invalidTo_failure() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("invalidToInternship.json"),
+                        JsonAdaptedInternship.class);
+        assertThrows(IllegalValueException.class,
+                Time.MESSAGE_CONSTRAINTS,
+                jsonAdaptedInternship.get()::toModelType);
+    }
+
+    @Test
+    public void loadToModel_invalidFromTo_failure() throws DataConversionException {
+        Optional<JsonAdaptedInternship> jsonAdaptedInternship =
+                readJsonFile(addToTestDataPathIfNotNull("invalidFromToInternship.json"),
+                        JsonAdaptedInternship.class);
+        assertThrows(IllegalValueException.class,
+                "The \"to\" field must not precede the \"from\" field.",
```

Comment:
might want to change this to be consistent with the error shown when adding item? It was 'from' cannot be later than 'to' field.